### PR TITLE
update_data_tables

### PR DESCRIPTION
**Pull request by EricBelongea**
## What kind of change does this PR introduce?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Small fix (Corrected a typo, removed a comment, etc.)

## Issue/feature/bug description: 

This will update the Breads and Countries table to have..
- Breads
   - imageUrl
   - ingredients
   - instructions 
 - Countries
    - imageUrl
    - overview
    - culinary
  
This fixes and adjusts all tests, models, and serializers for this data table change. 

All tests are passing and JSON is formatted how we wanted it.

## Checklist before requesting a review
- [x] I ran the code locally and verified that there are no visible errors.
- [x] I have performed a self-review of my code
- [x] I have refactored my code before this push
- [x] The commit message follows our guidelines
- [x] My code follows the code style of this project.

### CODE REVIEW

**Comment by memcmahon** [comment]
Code looks good - though I would be a bit careful on adding new features and refactors on the same PR.  My thought here is that a new feature might cause issues and need to rolled back, and then you'll lose your refactoring!  Likely not an issue that would come up on this project, but something to think about in larger code-bases :)

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Api::V1::UsersController < ApplicationController
+  def create
+    user = User.new(user_params)
+
+    if User.exists?(email: user.email)
```
Comment (coryrpow) [inline app/controllers/api/v1/users_controller.rb:5]: Got a if for checking for email uniqueness and password to match.

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+  end
+
+  def password_error(user)
+    {
```
Comment (coryrpow) [inline app/controllers/api/v1/users_controller.rb:23]: Got two simple error handling methods that could be refactored in to the error serializer, but work for now.

File: app/serializers/country_bread_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  def serialized_response
+    {
+      country: {
+        data: {
+          id: @country.id,
+          attributes: {
+            name: @country.name,
+            overview: @country.overview,
+            culinary: @country.culinary,
```
Comment (coryrpow) [inline app/serializers/country_bread_serializer.rb:15]: What a fancy handrolled JSON response! Also removes those relationship hashes that weren't needed for FE.

File: db/migrate/20240209035322_update_ingredients_in_breads.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class UpdateIngredientsInBreads < ActiveRecord::Migration[7.0]
+  def up
+    execute <<-SQL
+      ALTER TABLE breads ALTER COLUMN ingredients TYPE text[] USING ARRAY[ingredients];
```
Comment (coryrpow) [inline db/migrate/20240209035322_update_ingredients_in_breads.rb:4]: Still wrapping my brain around how these work, but stoked that we figured out how to get arrays in a schema! Pretty sick.

**Review by coryrpow — COMMENTED** [review]
Whew, lots of new stuff, but a lot of it is reformatting to work with the new needed keys for country and breads so the FE has access to more manageable elements, rather than one giant brick of text. The array were an interesting unforeseen challenge, but cool to know how to implement those in any Rails app going forward.

**Pull request merged**